iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -556,6 +556,11 @@ async Task OpenAsyncInternal(CancellationToken cancellationToken)
                 });
             }
 
+            if (this.channel == null)
```

Comment:
When would this happen?

Reply:
This happens if CreateChannelFactory iterates through all the provided IP addresses passed in and it receives ConnectExceptions for all of them. The method would just return a null channel object that is saved into this.channel

Reply:
Argh... this is the 3rd or 4th time we have issues around this area (@abhipsaMisra @varunpuranik / cc @prmathur-microsoft ).

We really need the e2e test and dotNetty integration test to ensure we're getting it right on both Windows and Linux. Please also include @nayato (Max) to review that portion of the code since this issue was caused by a wrong assumption about dotNetty.